boards/cc2538dk/include/periph_conf.h

```diff
@@ -59,33 +59,29 @@ static const timer_conf_t timer_config[] = {
 /** @} */
 
 /**
- * @name UART configuration
+ * @name    UART configuration
  * @{
  */
-#define UART_NUMOF          1
+static const uart_conf_t uart_config[] = {
+    {    /* Virtual COM Port */
+        .dev      = (cc2538_uart_t *)0x4000c000,
```

Comment:
I'd rather not have such address directly in the config, there should be defines for that. Or use numbers to index the UART used (i.e., `0`, `1`) and have a helper function in the driver to retrieve the corresponding `cc2538_uart_t`. See the timer implementation for instance, which has `static inline cc2538_gptimer_t *dev(tim_t Tim)` function.

Reply:
ok cool - do you have any advice about consistency - i.e. which of those two approaches is more widely implemented? In the sam0common drivers (IIRC) there's an implementation that's half way between, as well, where there's a helper function to retrieve the appropriate entry in the config array, but this doesn't solve the above problem.

Reply:
IMHO this is not about which approach is implemented more often, but a matter of efficiency: using a define with the UART base addresses here is more efficient than using an integer index and decoding it to the base address in the driver's c-file... So I see no reason for using an integer index here.